### DISCUSSION TITLE
Max body size

### DISCUSSION THREAD

**Discussion by vminfant** [Neoteroi/BlackSheep#381] [Q&A]
Hi,  Is there any way to use Max body size ? Also, if it is possible to use different max body size value for different endpoints?

**Comment by RobertoPrevato** [comment]
Hi @vminfant 
Thank You for a very good question. I wanted to verify before replying. Unfortunately `blacksheep` is lacking here:
- there is no way to validate the maximum body size globally (I thought `uvicorn` and other ASGI servers had this feature, but they do not either) - I included this feature initially when my web framework was not ASGI compatible, but later I removed it
- there is no way to validate the maximum body size completely in request handlers either

Currently it is only possible to validate the maximum body size efficiently when using the `request.stream()` async generator. I published an example for this here in [_BlackSheep Examples_](https://github.com/Neoteroi/BlackSheep-Examples/tree/main/max-body-size).

I created an issue https://github.com/Neoteroi/BlackSheep/issues/383 to look into this. The good news is that adding support for validating the maximum body size globally is easy, it is only necessary to do small changes to the `ASGIContent` class, or maybe wrapping the `receive` object received from the ASGI server. Controlling the maximum body size on a request handler basis is certainly more challenging. I prefer looking into this as changes can affect the performance of the web server.

**Reply by vminfant** [reply]
Hi @RobertoPrevato ... Great! Thank you for looking into this. 

Having a max body size on request handlers will be much more helpful. 

Thank you again for the examples.
